src/ralph/virtual/forms.py

```diff
@@ -0,0 +1,29 @@
+import json
+
+from django import forms
+from django.utils.translation import ugettext_lazy as _
+
+from ralph.admin.mixins import RalphAdminFormMixin
+from ralph.virtual.models import CloudProvider
+
+
+class CloudProviderForm(RalphAdminFormMixin, forms.ModelForm):
+    class Meta:
+        model = CloudProvider
+        fields = (
+            'name', 'cloud_sync_enabled', 'cloud_sync_driver', 'client_config'
+        )
+
+    def clean_client_config(self):
```

Comment:
I would consider using JSON field from django.extras package

Reply:
Makes sense. However, I will have to keep the custom form because that class does not implement any validation.